docs/source/app-dev/services.rst

```diff
@@ -68,16 +71,16 @@ For full details, see :ref:`the proto documentation for the service <com.daml.le
 Command deduplication
 ---------------------
 
-The command submission service deduplicates submitted commands based on the submitting :ref:`party <com.daml.ledger.api.v1.Commands.party>` and :ref:`command ID <com.daml.ledger.api.v1.Commands.command_id>`:
+The command submission service deduplicates submitted commands based on their change ID:
 
-- Applications can provide a :ref:`deduplication time <com.daml.ledger.api.v1.Commands.deduplication_time>` for each command. If this parameter is not set, the default maximum deduplication time is used.
-- A command submission is considered a duplicate submission if the ledger server receives the command within the deduplication time of a previous command with the same command ID from the same submitting party.
-- Duplicate command submissions will be ignored until either the deduplication time of the original command has elapsed or the original submission was rejected (i.e. the command failed and resulted in a rejected transaction), whichever comes first.
+- Applications can provide a :ref:`deduplication duration <com.daml.ledger.api.v1.Commands.deduplication_duration>` for each command. If this parameter is not set, the default maximum deduplication period is used.
+- A command submission is considered a duplicate submission if the ledger server receives the command within the deduplication period of a previous command with the same application ID and command ID from the same submitting party.
```

Comment:
That's not really what's happening in Canton any more. Canton uses the new idea of backward-looking dedup, i.e., the dedup period of the later submission counts, not the one of the accepted submission. I'm not sure what ledger implementations currently still use the forward-looking semantics; kvutils has been changed so that it always uses the max dedup time, which makes the two indistinguishable.

Reply:
> I'm not sure what ledger implementations currently still use the forward-looking semantics

I'm not sure either, perhaps @bame-da knows more. In addition to that, I'm not sure if the end state is having the backward-looking semantics specified as part of the API, of if the precise behavior should be left unspecified at the API level.

Reply:
> the end state is having the backward-looking semantics specified as part of the API

Yes. That's the desired end state once we have all officially supported ledger implementations moved over.

Reply:
How about a more general wording until we ensure that all ledgers comply with the new semantics:

```
- A command submission is considered a duplicate submission if the ledger API server is aware of another command within the deduplication period and with the same :ref`change ID <change-id>`.
 ```

Or, we could expand on the new semantics; I think you @bame-da can probably help deciding here.

Reply:
I like your unspecific wording here. It's always fine to make stronger promises in later releases.

Reply:
Alright, 4792a0a34be0a857ad7285ffda1d5d950a8c11a3. @bame-da (or someone else from product): I'll also wait for your feedback/stamp before merging.